ui/pages/asset/components/token-asset.tsx

```diff
@@ -28,7 +29,12 @@ const TokenAsset = ({ token, chainId }: { token: Token; chainId: Hex }) => {
   const { address, symbol, isERC721 } = token;
 
   const tokenList = useSelector(getTokenList);
-  const rpcPrefs = useSelector(getRpcPrefsForCurrentProvider);
+  const allNetworks: {
+    [key: `0x${string}`]: NetworkConfiguration;
+  } = useSelector(getAllEnabledNetworks);
+  // get the correct rpc url for the current token
+  const currentTokenBlockExplorer =
+    allNetworks[chainId]?.blockExplorerUrls?.[0];
```

Comment:
We should use the designated block explorer here instead of relying on the one at index 0. It should be implemented like this:
```
allNetworks[chainId]?.blockExplorerUrls[allNetworks[chainId]?.defaultBlockExplorerUrlIndex]
```
Since block explorer configuration is optional, relying on index 0 could lead to potential bugs if no block explorer is defined or the structure is missing. This approach ensures we use the intended block explorer safely.

Reply:
This actually got caught in an e2e test. I can address.